File: Application/app/src/main/java/com/example/sage/foodsharebook/Splash.java
package com.example.sage.foodsharebook;

import android.content.Intent;
import android.content.SharedPreferences;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

import com.example.sage.foodsharebook.Dishes.FoodListActivity;
import com.example.sage.foodsharebook.Users.LoginActivity;
import static com.example.sage.foodsharebook.Config.Constants.*;

public class Splash extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        SharedPreferences prefs = getApplicationContext().getSharedPreferences(SHARED_PREFERENCES_NAME,0);
        super.onCreate(savedInstanceState);
        String token = prefs.getString(USER_TOKEN,null);
        Intent intent;
        if (token == null){ intent = new Intent(this,LoginActivity.class); }
        else{ intent = new Intent(this, FoodListActivity.class); }
        startActivity(intent);
        finish();
    }
}

File: Application/app/src/main/java/com/example/sage/foodsharebook/Splash.kt
package com.example.sage.foodsharebook

import android.content.Intent
import android.content.SharedPreferences
import android.support.v7.app.AppCompatActivity
import android.os.Bundle

import com.example.sage.foodsharebook.Dishes.FoodListActivity
import com.example.sage.foodsharebook.Users.LoginActivity

import com.example.sage.foodsharebook.Config.Constants.SHARED_PREFERENCES_NAME
import com.example.sage.foodsharebook.Config.Constants.USER_TOKEN

class Splash : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        val prefs = applicationContext.getSharedPreferences(SHARED_PREFERENCES_NAME, 0)
        super.onCreate(savedInstanceState)
        val token = prefs.getString(USER_TOKEN, null)
        val intent: Intent
        if (token == null) {
            intent = Intent(this, LoginActivity::class.java)
        } else {
            intent = Intent(this, FoodListActivity::class.java)
        }
        startActivity(intent)
        finish()
    }
}
